### PR TITLE
Implement API action bulk:updateByQuery #51

### PR DESCRIPTION
**Pull request by CustomEntity**
Implement the API action bulk:updateByQuery

### How should this be manually tested?

Just call `BulkController#updateByQuery` method
 
Closes https://github.com/kuzzleio/sdk-jvm/issues/51

### CODE REVIEW

**Review by Shiranuit — APPROVED** [review]
Nice Job 👍

File: doc/1/controllers/bulk/update-by-query/index.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+<br/>
+
+| Argument         | Type                           | Description                                                                    |
+|------------------|--------------------------------|--------------------------------------------------------------------------------|
+| `index`          | <pre>String</pre>              | Index name                                                                     |
+| `collection`     | <pre>String</pre>              | Collection name                                                                |
+| `searchQuery`    | <pre>Map<String, Object></pre> | JSON representing the query to match                                           |
+| `changesQuery`   | <pre>Map<String, Object></pre> | JSON representing the query of the changes to apply                            |
```
Comment (Aschen) [inline doc/1/controllers/bulk/update-by-query/index.md:43]: This is not a query but simply a list of changes. It should be named `changes`

File: doc/1/controllers/bulk/update-by-query/index.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+| `waitForRefresh` | <pre>Bool</pre><br>(`false`)   | If set to true, Kuzzle will not respond until the delete documents are indexed |
+
+
+## Return
+
+An Int containing the number of updated documents.
```
Comment (Aschen) [inline doc/1/controllers/bulk/update-by-query/index.md:49]: Use code formatting for anything coming from the code
```suggestion
An `Int` containing the number of updated documents.
```

File: doc/1/controllers/bulk/update-by-query/snippets/update-by-query-kotlin.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+val searchQuery: Map<String, Any?> = HashMap<String, Any?>().apply {
+  put("query", HashMap<String, Any?>().apply {
+    put("term", HashMap<String, Any?>().apply {
+      put("capacity", 7)
+    })
+  })
+}
+
+val changesQuery: Map<String, Any?> = HashMap<String, Any?>().apply {
```
Comment (Aschen) [inline doc/1/controllers/bulk/update-by-query/snippets/update-by-query-kotlin.kt:9]: `changesQuery` should be renamed to `changes`

File: src/main/kotlin/io/kuzzle/sdk/controllers/BulkController.kt [hunk 4]
```diff
@@ -32,6 +32,33 @@ class BulkController(kuzzle: Kuzzle) : BaseController(kuzzle) {
             .thenApplyAsync { response -> (response.result as Map<String?, Any?>)["deleted"] as Int }
     }
 
+    @JvmOverloads
+    fun updateByQuery(
+        index: String,
+        collection: String,
+        searchQuery: Map<String, Any?>,
+        changesQuery: Map<String, Any?>,
```
Comment (Aschen) [inline src/main/kotlin/io/kuzzle/sdk/controllers/BulkController.kt:40]: Same comment about naming

File: doc/1/controllers/bulk/update-by-query/index.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+:::: tabs
+::: tab Java
+
+## Arguments
+
+```java
+public CompletableFuture<Int> updateByQuery(
+  String index,
+  String collection,
+  Map<String, Object> searchQuery,
```
Comment (Aschen) [inline doc/1/controllers/bulk/update-by-query/index.md:23]: Why are you not using the type defined in the code ? (`Any?`)

**Review by Aschen — CHANGES_REQUESTED** [review]

**Comment by Aschen** [comment]
@CustomEntity Are you planning to finish this?

**Review by Aschen — APPROVED** [review]

**Pull request merged**